views/layout.erb

```diff
@@ -0,0 +1,22 @@
+<html>
+
+  <body>
+  <header>
+  
+  <% if session[:user_id] != nil %>
```

Comment:
Because you created the @user variable in the route you can simply say:

if @user
 *show the user stuff*
else
*show the non user stuff*
end